src/server.js

```diff
@@ -65,20 +88,57 @@ const handleRender = (req, res, data) => {
   }
 };
 
+// redirect to viewer mode of the space
+const handleSpaceViewerRender = (req, res) => {
+  const { id } = req.params;
+  if (!ObjectId.isValid(id)) {
```

Comment:
If you're throwing an error here, you should catch it with some middleware so that the server doesn't collapse.

Reply:
Are you doing that?

Reply:
No, it just fails. Maybe we can redirect the user to a 404 page. I'll open an issue with this.